graylog2-server/src/test/java/org/graylog2/indexer/MongoIndexSetRegistryTest.java

```diff
@@ -266,4 +274,93 @@ public void isManagedIndexWithUnmanagedIndexReturnsFalse() {
 
         assertThat(indexSetRegistry.isManagedIndex("index")).isFalse();
     }
+
+    @Test
+    public void isCurrentWriteIndexSingleIndexSetConfigHappyPath() {
+        final String idxName = "index";
+        final IndexSetConfig indexSetConfig = mock(IndexSetConfig.class);
+        final List<IndexSetConfig> indexSetConfigs = Collections.singletonList(indexSetConfig);
+        final MongoIndexSet indexSet = mock(MongoIndexSet.class);
+        when(mongoIndexSetFactory.create(indexSetConfig)).thenReturn(indexSet);
+        when(indexSetService.findAll()).thenReturn(indexSetConfigs);
+        when(indexSet.isManagedIndex(idxName)).thenReturn(true);
+        when(indexSet.getActiveWriteIndex()).thenReturn(idxName);
+        assertThat(indexSetRegistry.isCurrentWriteIndex(idxName)).isTrue();
+
+        verify(indexSet).isManagedIndex(idxName);
+        verify(indexSet).getActiveWriteIndex();
+        verifyNoMoreInteractions(indexSetConfig, indexSet);
+    }
+
+    @Test
+    public void isCurrentWriteIndexSingleIndexSetConfigNoneFound() {
+        final String idxName = "index";
+        final IndexSetConfig indexSetConfig = mock(IndexSetConfig.class);
+        final List<IndexSetConfig> indexSetConfigs = Collections.singletonList(indexSetConfig);
+        final MongoIndexSet indexSet = mock(MongoIndexSet.class);
+        when(mongoIndexSetFactory.create(indexSetConfig)).thenReturn(indexSet);
+        when(indexSetService.findAll()).thenReturn(indexSetConfigs);
+        lenient().when(indexSet.getActiveWriteIndex()).thenReturn("non_existing_index");
+        assertThat(indexSetRegistry.isCurrentWriteIndex(idxName)).isFalse();
+
+        verify(indexSet).isManagedIndex(idxName);
+        verify(indexSetService).findAll();
+        verifyNoMoreInteractions(indexSetConfig, indexSet);
+    }
+
+    @Test
+    public void isCurrentWriteIndex500IndexSetConfigNoneFoundDueToMismatchingIndexName() {
+        final String idxName = "index";
+        final int noOfIndices = 500;
```

Comment:
Why 500? Seems like any number >1 would be pretty much equivalent for testing.
Test runtime is a big driver of our overall build time, so we want to keep that as low as possible (without sacrificing coverage).

Reply:
The number is from the original issue. But you're right, the problem can also be tested with only two indices.